### PR TITLE
Adds azd installation for devcontainer definition

### PR DESCRIPTION
**Pull request by murdockcrc**
This PR adds a post create command to the devcontainer definition, so that the container spins up with `azd` already installed (which is a project dependency).

### CODE REVIEW

File: .devcontainer.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
   // },
 
   // Use 'postCreateCommand' to run commands after the container is created.
-  // "postCreateCommand": "yarn install"
+  "postCreateCommand": "curl -fsSL https://aka.ms/install-azd.sh | bash"
```
Comment (pamelafox) [inline .devcontainer.json:59]: Why not use the azd feature? That will take care of architecture issues so that it works on ARM machines.

Comment (murdockcrc) [reply .devcontainer.json:59]: That is true, there are issues with Macs with Apple Silicon. But how to get azd installed in the first place? How do we solve that? @pamelafox thx.

Comment (pamelafox) [reply .devcontainer.json:59]: Use the feature: https://github.com/Azure-Samples/azure-search-openai-demo/blob/e91496076f2b061b0d9f898d3bdc3b9b87633323/.devcontainer/devcontainer.json#L11

**Comment by murdockcrc** [comment]
Closing because the main branch on the fork has changed too much in between, will open a separate branch for this change.

**Pull request closed without merging**